packages/angular/cli/models/schematic-command.ts

```diff
@@ -304,20 +304,18 @@ export abstract class SchematicCommand<
       ...(await getSchematicDefaults(schematic.collection.name, schematic.name, getProjectName())),
       ...current,
     });
-    workflow.engineHost.registerOptionsTransform(defaultOptionTransform);
 
-    if (options.defaults) {
-      workflow.registry.addPreTransform(schema.transforms.addUndefinedDefaults);
-    } else {
-      workflow.registry.addPostTransform(schema.transforms.addUndefinedDefaults);
-    }
+    workflow.engineHost.registerOptionsTransform(defaultOptionTransform);
 
+    workflow.registry.addPreTransform(schema.transforms.addUndefinedDefaults);
     workflow.registry.addSmartDefaultProvider('projectName', getProjectName);
     workflow.registry.useXDeprecatedProvider(msg => this.logger.warn(msg));
 
     if (options.interactive !== false && isTTY()) {
       workflow.registry.usePromptProvider((definitions: Array<schema.PromptDefinition>) => {
-        const questions: inquirer.QuestionCollection = definitions.map(definition => {
+        const questions: inquirer.QuestionCollection = definitions
+        .filter(definition => !options.defaults || definition.default == null)
```

Comment:
```suggestion
        .filter(definition => !options.defaults || definition.default == undefined)
```
I think this should be `undefined` since `null` is a valid default value.